src/crewai/cli/config.py

```diff
@@ -59,6 +73,26 @@ class Settings(BaseModel):
     )
     config_path: Path = Field(default=DEFAULT_CONFIG_PATH, frozen=True, exclude=True)
 
+    oauth2_provider: str = Field(
+        description="OAuth2 provider used for authentication (e.g., workos, okta, auth0).",
+        default=DEFAULT_CLI_SETTINGS["oauth2_provider"]
+    )
```

Comment:
Something just occurred to me: not sure if we want to babysit the user _that much_, but would it be possible to add allowed values to this field?

Right now the only possible values are `["workos", "auth0", "okta"]`.

If user tries `crewai config set oauth2_provider blabla` we would then raise an error saying that this is not a valid value for the field.

Reply:
sure, will do

Reply:
I'm facing some issues to do that due config set is using `setattr()`. So, I will make it in another PR